sdk/eventgrid/eventgrid/src/eventGridClient.ts

```diff
@@ -140,10 +163,25 @@ export class EventGridPublisherClient<T extends InputSchema> {
           );
         }
         case "CloudEvent": {
+          // The underlying REST API expects a header named `aeg-channel-name`, and so the generated client
+          // expects that options bag has a property called `aegChannelName`, where as we expose it with the
+          // friendlier name "channelName". Fix up the impedence mismatch here
+          const { channelName, ...sendOptions } = {
+            ...(updatedOptions as CloudEventSendOptions),
+          };
+
+          const generatedClientSendOptions: GeneratedClientPublishCloudEventEventsOptionalParams = {
+            ...sendOptions,
```

Comment:
is the spread operator necessary here? I think by using the rest params in the above it already has created a new object, so you can just pass `sendOptions` directly?

Reply:
@xirzec , Some of this is just me not being the best at TypeScript and trying to minimize the amount of `as`'s that I throw at the problem.  It felt like the destructing to pick apart the `channelName` from the rest of the `SendOptions` was the right thing to do here.  

You are correct that the destructing the spread is a logical no-op, but I found doing that would allow me to end up with something where I didn't have abuse `as`.

I could do something like this instead:

```ts
          // The underlying REST API expects a header named `aeg-channel-name`, and so the generated client
          // expects that options bag has a property called `aegChannelName`, where as we expose it with the
          // friendlier name "channelName". Fix up the impedence mismatch here
          const { channelName, ...sendOptions } = {
            ...(updatedOptions as CloudEventSendOptions),
          };

          if (channelName) {
            (sendOptions as GeneratedClientPublishCloudEventEventsOptionalParams).aegChannelName =
              channelName;
          }

          return this.client.publishCloudEventEvents(
            this.endpointUrl,
            (events as InputSchemaToInputTypeMap["CloudEvent"][]).map(convertCloudEventToModelType),
            sendOptions
          );
```

Would that be better?  Is there some typescript magic I can throw at my destructuring to tell that it should type `sendOptions` here as `GeneratedClientPublishCloudEventEventsOptionalParams`?  The shapes line up.